config/routes.rb

```diff
@@ -1,6 +1,9 @@
 # frozen_string_literal: true
 
 Rails.application.routes.draw do
-  mount_devise_token_auth_for 'User', at: 'auth'
+  mount_devise_token_auth_for 'User', at: 'auth', controllers: {
+    registrations: 'override/registrations',
+    sessions: 'override/sessions'
+  }
```

Comment:
So just you know, `devise` has [`parent_controller`](https://github.com/heartcombo/devise/blob/5d5636f03ac19e8188d99c044d4b5e90124313af/lib/devise.rb#L230-L234)

You can create base `DeviseParentController` or something like that and set it there - that way you can have simple base for all devise controllers that will not inherit any changes, like enforced authentication, maybe some other overrides, additional settings etc. It for sure not required at this step, as those overrides are doing their job properly, but this is some other option.